### DISCUSSION TITLE
govc and vim-cmd / hostsvc

### DISCUSSION THREAD

**Discussion by ralfgro** [vmware/govmomi#3595] [Community Help] [answered]
I noticed that vim-cmd is not available for govc. Usually host.esxcli is working for me but I've noticed that not everything is available for esxcli.

Examples:

vim-cmd hostsvc/advopt/update ScratchConfig.ConfiguredScratchLocation string $scratchdirectory

There seems to be no esxcli command to set the general scratch location, there is something for a persistent log location, but not ScratchConfig.ConfiguredScratchLocation. Or...

vim-cmd hostsvc/firmware/sync_config
vim-cmd hostsvc/firmware/backup_config

There seem not be the anything for hostsvc in esxcli. Is there any alternative for this in govc?

**Comment by dougm** [comment, marked as answer]
iirc, `vim-cmd` is its own cli that uses the same APIs govc. I don't know vim-cmd well, but looks like govc host.option.{ls,set} can do what you need:
```console
% govc host.option.ls -host $host ScratchConfig.ConfiguredScratchLocation
ScratchConfig.ConfiguredScratchLocation:  /vmfs/volumes/67104522-e4406397-678b-02000da04ab4

% govc host.option.set -host $host ScratchConfig.ConfiguredScratchLocation /tmp

% govc host.option.ls -host $host ScratchConfig.ConfiguredScratchLocation
ScratchConfig.ConfiguredScratchLocation:  /tmp
```

**Reply by ralfgro** [reply]
Thx, that seems to work!
